### PR TITLE
Remove parsing

### LINKED ISSUE #254: Discrepancy between local docker image & latest web calculations
There appears to be a mathematical error between docker image ` 1.1.1.` (downloaded last year) and the latest web version (Oct 7th, 24). 

Inputting the same concentrations and reactions between both models results in different results with the web version appearing to be inaccurate by an order of 10^3. 

NOx chemistry reaction:
* species
  * N2,  
  * O2, 
  * NO, 
  * NO2, 
  * O, 
  * O3
* Rxn
  * NO2 -> NO + O (photolysis)
    * MUSICA: NO2photo
    * Rxn Conditions: NO2photo = 0.01
  * O + O2 + M -> O3 + M (Arrhenius) 
    * A = 6e-34
  * NO + O3  ->  NO2 + O2
    * A = 1.8e-14
* concentrations
  * N2 = 2e19 molecules/cm3
  * O2 = 5e18 molecules/cm3
  * NO = 0
  * NO2 = 1e12 molecules/cm3 
  * O3 = 0

We are expecting this graph: 
image

But, those results only occur if the concentrations are bumped up by an order of 10^3 to: 
* concentrations
  * N2 = 2e22 molecules/cm3
  * O2 = 5e21 molecules/cm3
  * NO2 = 1e15 molecules/cm3

### PR DESCRIPTION
**Pull request by K20shores**
We don't need to parse the mechanism information in python. This removes that work, as well as simplifies some of the parsing and setting of conditions. Also, unit conversions from anything in cm3 to m3 is corrected.

Closes #254

### CODE REVIEW

**Review by mattldawson — DISMISSED** [review]

File: src/acom_music_box/conditions.py [hunk 1]
```diff
@@ -43,8 +34,8 @@ def __init__(
         """
         self.pressure = pressure
         self.temperature = temperature
-        self.species_concentrations = species_concentrations if species_concentrations is not None else []
-        self.reaction_rates = reaction_rates if reaction_rates is not None else []
+        self.species_concentrations = species_concentrations if species_concentrations is not None else {}
```
Comment (boulderdaze) [inline src/acom_music_box/conditions.py:37]: Could you update the description in `line 32-33` to reflect the changes in the data container? 

```
species_concentrations (List[SpeciesConcentration]): A list of species concentrations. Default is an empty list.
reaction_rates (List[ReactionRate]): A list of reaction rates. Default is an empty list.
```

File: src/acom_music_box/conditions.py [hunk 2]
```diff
@@ -117,9 +108,7 @@ def from_UI_JSON(self, UI_JSON, species_list, reaction_list):
     def from_config_JSON(
             self,
             path_to_json,
-            config_JSON,
-            species_list,
-            reaction_list):
+            object):
```
Comment (boulderdaze) [inline src/acom_music_box/conditions.py:111]: Could you update the descriptions for the arguments? `line 121-122` is no longer needed and we can add a description for `object`.

File: src/acom_music_box/conditions.py [hunk 3]
```diff
@@ -136,67 +125,45 @@ def from_config_JSON(
...
         return self(
             pressure,
             temperature,
-            species_concentrations,
+            initial_concentrations,
             reaction_rates)
 
     @classmethod
-    def read_initial_rates_from_file(self, file_path, reaction_list):
+    def read_initial_rates_from_file(cls, file_path):
```
Comment (boulderdaze) [inline src/acom_music_box/conditions.py:166]: Could you update the arguments' descriptions?

File: src/acom_music_box/conditions.py [hunk 4]
```diff
@@ -211,24 +178,26 @@ def read_initial_rates_from_file(self, file_path, reaction_list):
...
+                reaction_type, label, units = parts
+            elif len(parts) == 2:
+                reaction_type, label = parts
+            else:
+                logger.error(f"Unexpected format in key: {key}")
+            rate_name = f'{reaction_type}.{label}'
+            if rate_name in reaction_rates:
+                raise ValueError(f"Duplicate reaction rate found: {rate_name}")
+            reaction_rates[rate_name] = df.iloc[0][key]
+
```
Comment (boulderdaze) [inline src/acom_music_box/conditions.py:200]: What is the reason to add unexpected-format keys into `reaction rates` dict? My impression based on the logging level is that it's not valid and should be thrown away

File: src/acom_music_box/music_box.py [hunk 5]
```diff
@@ -290,38 +231,23 @@ def loadJson(self, path_to_json):
...
-            self.initial_conditions = Conditions.from_config_JSON(
-                path_to_json, data, self.species_list, self.reaction_list)
+            self.initial_conditions = Conditions.from_config_JSON(path_to_json, data)
 
             # Set initial conditions
-            self.evolving_conditions = EvolvingConditions.from_config_JSON(
-                path_to_json, data, self.species_list, self.reaction_list)
-
-        self.check_config(os.path.join(os.getcwd(), path_to_json))
+            self.evolving_conditions = EvolvingConditions.from_config_JSON(path_to_json, data)
```
Comment (boulderdaze) [inline src/acom_music_box/music_box.py:242]: Could you update the comment (line 241)?

```
# Set evolving conditions
```

File: src/acom_music_box/music_box.py [hunk 6]
```diff
@@ -290,38 +231,23 @@ def loadJson(self, path_to_json):
...
-            camp_path,
-            musica.micmsolver.rosenbrock,
-            1)
+        # Initalize the musica solver
+        self.solver = musica.create_solver(camp_path, musica.micmsolver.rosenbrock, 1)
 
-    @classmethod
-    def order_reaction_rates(self, curr_conditions, rate_constant_ordering):
+    @staticmethod
+    def order_reaction_rates(curr_conditions, rate_constant_ordering):
```
Comment (boulderdaze) [inline src/acom_music_box/music_box.py:250]: Could you update the description for the arguments for this function?

File: tests/integration/test_chapman.py [hunk 7]
```diff
@@ -14,50 +14,24 @@ def test_run(self):
...
-                    float(model_output_concs[i][j]),
-                    float(test_output_concs[i][j]),
-                    rel_tol=1e-7,
-                    abs_tol=1e-15,
-                ), f"Arrays differ at index ({i}, {j}) for species {concs_to_test[j]}"
+                    _model[column],
+                    _expected[column],
+                    rel_tol=1e-19,
+                    abs_tol=1e-16,
+                ), f"Model results differ from expected for row index {_model_index} for species {column}"
```
Comment (boulderdaze) [inline tests/integration/test_chapman.py:34]: Just curious, why do `_model` and `_expected` variables have a leading underscore?

File: src/acom_music_box/conditions.py (hunk 1 above)
Comment (K20shores) [reply src/acom_music_box/conditions.py:37]: Ah, yes. Thanks for catching that!

File: src/acom_music_box/conditions.py (hunk 2 above)
Comment (K20shores) [reply src/acom_music_box/conditions.py:111]: Yes

File: tests/integration/test_chapman.py (hunk 7 above)
Comment (K20shores) [reply tests/integration/test_chapman.py:34]: I declared model and expected above and these underscores are used as loop variables

**Review by boulderdaze — APPROVED** [review]
Thank you for addressing the comments!

**Pull request merged**